### ISSUE TITLE
Intermittent issues with large responses

### ISSUE THREAD

**Issue by thefosk** [Mashape/kong#689]
For larger responses (around 2MB), the output is getting cut off arbitrarily in the middle of the JSON. Seems to only happen when requesting non-gzip/deflate content, though i can't rule it out from happening for gzipped/deflated content. (kong 0.4.2). returned headers seem ok:

```
HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Cache-Control: max-age=0, no-cache, private
Content-Type: application/json
Date: Mon, 02 Nov 2015 16:12:32 GMT
Etag: 45468d12bd36453dc9bf54f26a0a5ce3
Server: Apache
Strict-Transport-Security: max-age=63072000; includeSubdomains; preload
Vary: Accept-Encoding,User-Agent
Via: kong/0.4.2
X-RateLimit-Limit-day: 100000
X-RateLimit-Limit-hour: 10000
X-RateLimit-Limit-second: 25
X-RateLimit-Remaining-day: 99991
X-RateLimit-Remaining-hour: 9993
X-RateLimit-Remaining-second: 24
transfer-encoding: chunked
Connection: keep-alive
```

My first guess is that it seems to be an issue with chunked responses.

Plugins enabled are: filelog, cors, mashape-analytics, rate limiting, and request size limiting.

**Comment by thibaultCha** [comment]
Probably has to do with the nginx proxy configured directives: http://nginx.org/en/docs/http/ngx_http_proxy_module.html

**Comment by thefosk** [comment]
@lucian303 to help me debugging this issue, how much time does it usually take to return the response? I am wondering if it has to do with some timeout setting.

**Comment by thefosk** [comment]
As @thibaultCha suggested, it may also be related to another issue:

If the response is too big and can't be held in the `proxy_buffers`, the rest of the response gets written in the `proxy_temp` directory. If for some reason nginx fails to store the response (not enough disk space, for example), then the response is truncated. This also happens if the size of the response exceeds the [`proxy_max_temp_file_size`](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_max_temp_file_size) property.

@lucian303, one more question - are you using the default Kong settings or did you change the nginx configuration?

**Comment by thefosk** [comment]
I tried to replicate this issue with both a chunked and un-chunked response of 6MB, with no success (using Kong v0.5.2).

@lucian303 can you try with the latest version of Kong?
<hr>
If somebody else experiences this issue, please comment on this issue.

**Comment by lucian303** [comment]
@thefosk I was getting this with the default Kong settings and nginx settings. The time for the response is about the same whether it's coming fully through or getting cut off (no obvious timeout). I'm not sure if the same issue shows up in 0.5.2 or not as I haven't had a chance to fully test it out, but in the little bit of testing I did manage, I did not see it in 0.5.2 (I'm hoping it's ok in 0.5.2, but my testing wasn't thorough to be able to say one way or another).
